src/schemapack/_internals/validation/_main.py

```diff
@@ -67,22 +67,14 @@ def _create_plugins_by_class(
 def _create_plugins_by_class(
     *,
     schemapack: SchemaPack,
-    plugin_classes: Union[
-        list[type[ClassValidationPlugin]],
-        list[type[ResourceValidationPlugin]],
-        list[Union[type[ClassValidationPlugin], type[ResourceValidationPlugin]]],
-    ],
-) -> Union[
-    dict[str, list[ClassValidationPlugin]],
-    dict[
-        str,
-        list[ResourceValidationPlugin],
-    ],
-    dict[
-        str,
-        list[Union[ResourceValidationPlugin, ClassValidationPlugin]],
-    ],
-]:
+    plugin_classes: list[type[ClassValidationPlugin]]
+    | list[type[ResourceValidationPlugin]]
+    | list[type[ClassValidationPlugin] | type[ResourceValidationPlugin]],
+) -> (
+    dict[str, list[ClassValidationPlugin]]
+    | dict[str, list[ResourceValidationPlugin]]
+    | dict[str, list[ResourceValidationPlugin | ClassValidationPlugin]]
+):
```

Comment:
Shouldn't the last case include the other both and this can be simplified?
Or does this run into some shenanigans with variance in that case?